File: library/src/main/java/com/trinity/recording/service/factory/PlayerServiceFactory.java
package com.trinity.recording.service.factory;

import android.os.Handler;
import com.trinity.recording.RecordingImplType;
import com.trinity.recording.service.PlayerService;
import com.trinity.recording.service.impl.AudioTrackPlayerServiceImpl;

public class PlayerServiceFactory {

	private static PlayerServiceFactory instance = new PlayerServiceFactory();

	private PlayerServiceFactory() {
	}

	public static PlayerServiceFactory getInstance() {
		return instance;
	}

	public PlayerService getPlayerService(final PlayerService.OnCompletionListener onComletionListener, RecordingImplType recordingImplType,
										  Handler mTimeHandler) {
		PlayerService result = null;
		switch (recordingImplType) {
		case ANDROID_PLATFORM:
			result = new AudioTrackPlayerServiceImpl() {
				@Override
				public void onCompletion() {
					onComletionListener.onCompletion();
				}
			};
			break;
		default:
			break;
		}
		result.setHandler(mTimeHandler);
		return result;
	}
}

File: library/src/main/java/com/trinity/recording/service/factory/PlayerServiceFactory.kt
package com.trinity.recording.service.factory

import com.trinity.recording.RecordingImplType
import com.trinity.recording.service.PlayerService
import com.trinity.recording.service.impl.AudioTrackPlayerServiceImpl

class PlayerServiceFactory private constructor() {

  fun getPlayerService(
    onComletionListener: PlayerService.OnCompletionListener,
    recordingImplType: RecordingImplType
  ): PlayerService? {
    var result: PlayerService? = null
    when (recordingImplType) {
      RecordingImplType.ANDROID_PLATFORM -> result = object : AudioTrackPlayerServiceImpl() {
        override fun onCompletion() {
          onComletionListener.onCompletion()
        }
      }
    }
    return result
  }

  companion object {
    @JvmStatic
    val instance = PlayerServiceFactory()
  }
}
